### ISSUE TITLE
Backfield Curve Unmixing

### ISSUE THREAD

**Issue by maxwellbrown** [PmagPy/RockmagPy-notebooks#13]
Develop MaxUnmix for python. I think Mike Volk may have done this. He did email me today, so I can reach out to him about this.

**Comment by Swanson-Hysell** [comment]
There is some code in PmagPy that @botaoxiongyong developed in this regard that is illustrated here:

<img width="941" alt="Screenshot 2024-02-14 at 8 48 11 AM" src="https://github.com/PmagPy/RockmagPy-notebooks/assets/4332322/72f5a688-a4bc-4461-8c93-3d0b4caf43b5">

The code that @botaoxiongyong developed is here: https://github.com/botaoxiongyong/pyIRM
and is implemented in this notebook: https://github.com/botaoxiongyong/pyIRM/blob/master/pyIRM/pyIRM.ipynb

@duserzym has experimented with approaches to do such unmixing in Python as well.

**Comment by maxwellbrown** [comment]
Very nice. Thanks, Nick and Jiabo @botaoxiongyong.

**Comment by duserzym** [comment]
Very cool functions! 

The way the best fit and the uncertainty bounds are determined are different than MaxUnmix. Worth adding more detailed description once added to RockmagPy!

**Comment by josh-feinberg** [comment]
Yes, we spent a lot of time discussing this when developing MaxUnmix and it was an issue for some of the reviewers too. Definitely want to add detailed descriptions of how data are being processed.

**Comment by Swanson-Hysell** [comment]
Our idea here is that we can develop a VSM_backfield.ipynb that:

- [ ] plots the data
- [ ] calculates Bcr
- [ ] calculates the derivative
- [ ] fit coercivity spectra

**Comment by duserzym** [comment]
I made an attempt with package lmfit committed at 77c5518. The method follows the example given here: https://lmfit.github.io/lmfit-py/model.html in lmfit which implements the method of https://www.astro.rug.nl/software/kapteyn/kmpfittutorial.html#confidence-and-prediction-intervals which is based on J. Wolberg, Data Analysis Using the Method of Least Squares, 2006, Springer. It does not require resampling and refitting of the data. I believe both Max Unmix and Jiabo's methods resample data to generate uncertainty regions. More info here: https://www.astro.rug.nl/software/kapteyn/kmpfittutorial.html#confidence-and-prediction-intervals

**Comment by duserzym** [comment]
@peat22 Hi Peat there is an issue with the exported backfield measurements in MagIC format that does not exist in the Excel format. 

If you take a look at columns `magn_mass` and `treat_dc_field` in the measurement table of my SSRM2022C MagIC export (https://github.com/PmagPy/RockmagPy-notebooks/tree/main/example_data/SSRM2022C), you will find that for some specimens, the values in these two fields sometimes go wrong in the middle of the experiments. The `magn_mass` will miss the negative signs, and jump to either very big values or very small values. I am guessing this has to do with counting character positions when parsing the VSM data in the IRMDB when exporting into MagIC format.

**Comment by duserzym** [comment]
backfield unmixing is now implemented and shown in https://github.com/PmagPy/RockmagPy-notebooks/tree/main/hysteresis_backfield_notebooks
